gobblin-api/src/main/java/gobblin/configuration/ConfigurationKeys.java

```diff
@@ -47,6 +47,13 @@
   public static final String STATE_STORE_FS_URI_KEY = "state.store.fs.uri";
   // Enable / disable state store
   public static final String STATE_STORE_ENABLED = "state.store.enabled";
+  // DB state store configuration
+  public static final String STATE_STORE_DB_ENABLED_KEY = "state.store.db.enabled";
```

Comment:
We should have a more generic way of instantiating the state store rather than just having two options: file system and mysql. We certainly want to be able to support other SQL stores (e.g. Oracle or posgres) or other types of stores (like Espresso).

I suggest we have a Factory based approach. Something along the lines of:

public interface StateStoreFactory {
   StateStore createStateStore(Config storeConfig);
}

Then we ca have a config property, say, state.store.type that takes a factory class or a human readable mnemonic. You can have a look at gobblin.http.HttpClientConfiguratorLoader.getConfiguratorClass(Optional<String>) for a pattern on how to support both syntaxes. createStateStore() can then just instantiate the factory class and create the store.